pkg/model/field.go

```diff
@@ -111,6 +114,41 @@ func (f *Field) IsReference() bool {
 	return trimmedGoType == "*ackv1alpha1.AWSResourceReferenceWrapper"
 }
 
+// GetReferenceFieldName returns the corresponding Reference field name
+// Reference field name is generated by removing the identifier suffix like
+// 'Id', 'Arn', 'Name' etc and adding 'Ref(s)' as suffix.
+func (f *Field) GetReferenceFieldName() names.Names {
+	if f.Names.Original == "" {
+		panic(fmt.Sprintf("field with empty name inside crd %s", f.CRD.Names.Original))
+	}
+	refNamePrefix := f.Names.Original
+	identifierSuffixes := []string{
+		"id", "ids", "Id", "Ids", "ID", "IDs", "IDS",
+		"Name", "Names", "NAME", "NAMEs", "NAMES",
+		"Arn", "Arns", "ARN", "ARNs", "ARNS",
+	}
+	for _, suffix := range identifierSuffixes {
+		if strings.HasSuffix(f.Names.Original, suffix) {
+			refNamePrefix = strings.TrimSuffix(refNamePrefix, suffix)
+			break
+		}
+	}
+	if refNamePrefix == "" {
+		panic("The corresponding field name for a reference field cannot" +
+			" be just 'id(s)' or 'arn(s)' or 'name(s)'. Current value: " +
+			f.Names.Original)
+	}
+	refName := refNamePrefix
+	// If the shape of corresponding field is a list, singularize the refNamePrefix
+	// and add Refs at the end
+	if f.ShapeRef != nil && f.ShapeRef.Shape != nil && f.ShapeRef.Shape.Type == "list" {
```

Comment:
yup, this should work perfectly for now. I don't know of any `map[string]*AWSResourceReferenceWrapper` fields (yet!) so I think we're good here.